<a href="https://colab.research.google.com/github/Medder001/ML-Final/blob/main/ML_Final_Kay%26Megan.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [20]:
from google.colab import drive
drive.mount("/content/drive", force_remount=True)
DRIVE_PATH = "/content/drive/MyDrive/archive"


Mounted at /content/drive


In [17]:
import os
import numpy as np
from PIL import Image
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.decomposition import PCA

In [28]:
def load_images(folder, label, img_size=(64, 64), max_images=1000):
  data = []
  filenames = os.listdir(folder)
  np.random.shuffle(filenames)
  count = 0
  for fname in os.listdir(folder):
    if count >= max_images:
      break
    try:
      img = Image.open(os.path.join(folder, fname)).convert('RGB')
      img = img.resize(img_size)
      pixels = np.array(img).flatten()
      data.append({'pixels': pixels, 'label':label})
      count += 1
    except Exception:
      pass
  return data

In [31]:
ai_data = load_images("/content/drive/MyDrive/archive/AiArtData/AiArtData", label=1)
real_data = load_images("/content/drive/MyDrive/archive/RealArt/RealArt", label = 0)

df = pd.DataFrame(ai_data + real_data)

X = np.stack(df['pixels'].values)
y = df['label'].values

print(f"Total instance: {len(df)}")
print(df['label'].value_counts())

size_mb = df['pixels'].apply(lambda x: x.nbytes).sum() / (1024 * 1024)
print(f"Loaded data size: {size_mb:.2f} MB")


X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42, stratify=y)

X_train = X_train / 225.0
X_test = X_train / 225.0

/usr/local/lib/python3.12/dist-packages/PIL/Image.py:1047: UserWarning: Palette images with Transparency expressed in bytes should be converted to RGBA images
  warnings.warn(


Total instance: 975
label
1    539
0    436
Name: count, dtype: int64
Loaded data size: 2925.00 MB


Our dataset is sampled for an AI-generated image detector ML model from Kaggle (https://www.kaggle.com/datasets/cashbowman/ai-generated-images-vs-real-images).
We sample up to 1,000 images per class with a result of 975 image instances in our dataset, specifically 539 AI-generated images and 436 real images. The images are resized to 64x64 and formatted as 'RGB'. From this, the calculated number of numeric attributes is 64 * 64 * 3 = 12,228 which are normalized to a continuous range of [0,1]. Additonally, any corrupted files are passed over.